textile/features.textile

```diff
@@ -2405,6 +2411,44 @@ class ReferenceExtras: // REX*
 class ClientInformation: // CR*
   +agents: Dict<String, String?> // CR2
   +agentIdentifier(additionalAgents: Dict<String, String?>?) => String // CR3; interface only offered by some libraries
+
+class BatchResult<T>
+  successCount: number // BAR2a
+  failureCount: number // BAR2b
+  results: [T] // BAR2c
+
+class BatchPublishSpec:
+  channels: [String] // BSP2a
+  messages: [Message] //BSP2b
+
+class BatchPublishSuccessResult:
+  channel: string // BPR2a
+  messageId: string // BPR2b
+
+class BatchPublishFailureResult:
+  channel: string // BPF2a
+  error: ErrorInfo // BPF2c
+
+class BatchPresenceSuccessResult:
+  channel: string // BGR2a
+  presence?: [PresenceMessage] // BGR2b
```

Comment:
@owenpearson @SimonWoolf is `presence` intentionally optional here? Ditto for whether the "if the request succeeded" in BGR2b is redundant.

Reply:
yeah you're right --  I guess it was originally written as BatchPublishResult with `channel & presence? & error?` and then later split into `{channel & presence} | {channel and error}` but we forgot to remove the optional mark. similarly in other places